pkg/asset/installconfig/clusterid.go

```diff
@@ -1,26 +1,50 @@
 package installconfig
 
 import (
+	"fmt"
+
 	"github.com/pborman/uuid"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
 
 	"github.com/openshift/installer/pkg/asset"
 )
 
+const (
+	// resource using HumanID usually have suffixes like `[-/_][a-z]{3,4}` eg. `_int`, `-ext` or `-ctlp`
+	maxNameLen = 32 - 5
+	randomLen  = 5
+	maxBaseLen = maxNameLen - randomLen - 1
+)
+
 // ClusterID is the unique ID of the cluster, immutable during the cluster's life
 type ClusterID struct {
-	ClusterID string
+	// UUID is a globally unique identifier.
+	UUID string
+
+	// HumanID is an identifier for the cluster that is more human friendly.
+	// This does not have
+	HumanID string
 }
 
 var _ asset.Asset = (*ClusterID)(nil)
 
-// Dependencies returns no dependencies.
+// Dependencies returns install-config.
 func (a *ClusterID) Dependencies() []asset.Asset {
-	return []asset.Asset{}
+	return []asset.Asset{
+		&InstallConfig{},
+	}
 }
 
-// Generate generates a new UUID
-func (a *ClusterID) Generate(asset.Parents) error {
-	a.ClusterID = uuid.New()
+// Generate generates a new ClusterID
+func (a *ClusterID) Generate(dep asset.Parents) error {
+	ica := &InstallConfig{}
+	dep.Get(ica)
+	base := ica.Config.ObjectMeta.Name
+	if len(base) > maxBaseLen {
+		base = base[:maxBaseLen]
```

Comment:
I think we want to trim sensitive characters, see #1089.

Reply:
Hmm, this is also imposing AWS-specific length and (eventually) charset restrictions on a nominally-platform-agnostic property.  Will this end up as a race to the bottom?  Maybe it should be a platform-specific asset?  Or do we expect this to be something almost all platforms will need?

Reply:
I considered this same thing with the PR that I put out earlier. My decision was that for the time being there is no reason to make the HumanID characteristics unique to the platform as we do not have any platforms currently that place more onerous restrictions than we face with ELBs from AWS. It should be easy enough to change to platform-specific HumanIDs in the future if and when they are beneficial.

Reply:
Yeah, we should be able to send cluste_ids to different platforms if that becomes a problem.
IMO this HumanID has enough constraints to work through most.

> I think we want to trim sensitive characters too, see #1089.

Will do.

Reply:
> > I think we want to trim sensitive characters too, see #1089.
> 
> Will do.

added here https://github.com/openshift/installer/pull/1280/commits/55b241004756c2b9636f6527196ed3bc77a6d094